src/cmd/chat_mode.rs

```diff
@@ -0,0 +1,215 @@
+use crate::{
+  claude_client::{ClaudeClient, ConversationId},
+  config::Config,
+  persistence::PersistentStore,
+};
+use serenity::{async_trait, model::channel::Message, prelude::Context};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::{error, info, instrument, warn};
+
+use super::MessageListener;
+
+pub struct ChatModeHandler {
+  claude_client: Arc<Mutex<ClaudeClient>>,
+}
+
+impl ChatModeHandler {
+  pub fn new(config: &Config, persistence: Arc<PersistentStore>) -> Self {
+    let claude_client = Arc::new(Mutex::new(ClaudeClient::new(config, persistence).expect(
+      "Failed to create Claude client - check configuration and Claude CLI availability",
+    )));
+
+    Self { claude_client }
+  }
+
+  fn extract_user_message(&self, ctx: &Context, msg: &Message) -> Option<String> {
+    let current_user_id = ctx.cache.current_user().id;
+    let bot_mention = format!("<@{}>", current_user_id);
+    let bot_mention_nick = format!("<@!{}>", current_user_id);
+
+    // Remove bot mentions from the message content
+    let content = msg
+      .content
+      .replace(&bot_mention, "")
+      .replace(&bot_mention_nick, "")
+      .trim()
+      .to_string();
+
+    Some(content).filter(|s| !s.is_empty())
+  }
+
+  async fn send_response(
+    &self,
+    ctx: &Context,
+    msg: &Message,
+    response: &str,
+  ) -> Result<(), String> {
+    const MAX_MESSAGE_LENGTH: usize = 1000;
+
+    if response.len() <= MAX_MESSAGE_LENGTH {
+      msg
+        .reply(&ctx.http, response)
+        .await
+        .map_err(|e| format!("Failed to send reply: {}", e))?;
+    } else {
+      let (emote_name, bot_nick) = {
+        let config = Config::global_instance().read().unwrap();
+        (
+          config.emote_name.clone(),
+          ctx.cache.current_user().name.clone(),
+        )
+      };
+      let snarky_response = format!(
+        "{} doesn't really feel like speaking that much {}",
+        bot_nick, emote_name
+      );
+      msg
+        .reply(&ctx.http, &snarky_response)
+        .await
+        .map_err(|e| format!("Failed to send snarky response: {}", e))?;
+    }
```

Comment:
Rather than checking a preconfigured max message length, attempt to send the response anyways and if there's an error reply with the snarky_response.